### DISCUSSION TITLE
booting for the first time

### DISCUSSION THREAD

**Discussion by ogg1e** [mebs/CM4-NAS#29] [Q&A] [answered]
So after I've flashed the board with the standard raspberry pi image using the imager tool, what should I expect for behaviour?

Do I set the usb jumper to master?

Should I see video via hdmi, or do I need to set that as output somehow?

Mine boots, and I see the activity light flash for a while , but I don't have video, and I don't see it picking up an IP address via dhcp on my router.

**Comment by ogg1e** [comment]
So I actually found that a pin on the HDMI port wasn't soldered right. So I do get video out over HDMI now, but it throws the error:

`'config failed hub doesn't have any ports! (err -19)'`

So is there certain config settings I need to change after flashing the OS to the eMMC? Any idea @mebs ?

I'm not trying to boot from usb, and I don't have any devices plugged into USB yet.

**Reply by ogg1e** [reply]
So here's a more detailed update:
I updated boot/config.txt to disable OTG and that got rid of the hub not having any ports
The boot sequence I see over hdmi is:

- the rainbow type boot screen
- the text error screen with the error : brcm-pcie link down (but that's ok.. I have no cards installed yet, and no rj45 cable connected)
- a quick flash of the white screen with a raspberry on it
- then nothing - the video turns off

The activity LED stays on and never blinks or shuts off at this point.

**Comment by ogg1e** [comment, marked as answer]
Turns out it was a bad power supply. Even though it reads 5v and 12v without load, and is "rated" at 2amp for 5v.

**Reply by mebs** [reply]
Glad you made it work! Sorry, I haven't been available to reply until now.
So you assembled the board yourself? How did it go? Happy to see someone else benefited from this :)

**Reply by ogg1e** [reply]
I did assemble the board myself.   I'm just trying to figure out the best solution for a proper power supply now that I know the one I have is bad.

**Reply by ogg1e** [reply]
And soldering the cm4 connectors wasn't fun.  I had the iron too hot and melted the plastic.  The rest was no problem.

**Reply by mebs** [reply]
Yeah that connector is the only challenge, especially now that stocks are low everywhere.
For the power supply, I just use a cheap 12V 10A switching PSU, and for the 5V rail I use this exact buck converter:
https://www.aliexpress.com/item/32810967274.html?spm=a2g0o.9042311.0.0.27424c4d4AB3Ne

**Reply by ogg1e** [reply]
what did you use for the panel mount barrel connector @mebs ?

**Reply by mebs** [reply]
Yes, I used a regular barrel female connector on the box. The ground is shared across the whole system and +12V serves the buck converter and the board itself.
Remember that there's no power management on the board so a good stable power supply is required.

**Reply by ogg1e** [reply]
I think I might buy this one to be safe: https://www.digikey.ca/en/products/detail/globtek-inc/TR9CE9200LCP-IM-R6B/8598052
